core/templates/dev/head/services/AssetsBackendApiService.js

```diff
@@ -0,0 +1,109 @@
+// Copyright 2017 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Service to serve as the interface for fetching and uploading
+ * assets from Google Cloud Storage.
+ */
+
+oppia.factory('AssetsBackendApiService', [
+  '$http', '$q', 'UrlInterpolationService',
+  function(
+      $http, $q, UrlInterpolationService) {
+    var GCS_AUDIO_DOWNLOAD_URL_TEMPLATE =
+      '/createhandler/audioupload/<exploration_id>';
+    var AUDIO_UPLOAD_URL_TEMPLATE = 
+      'https://storage.googleapis.com/oppia.resources/' +
```

Comment:
The string "oppia.resources" should not be hardcoded. It should reference the relevant constant in constants.js so that if we change its value in the backend, the value in the frontend also changes.